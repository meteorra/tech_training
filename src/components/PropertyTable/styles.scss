.property-table {
    background: url('./test.png') left top no-repeat;
}
