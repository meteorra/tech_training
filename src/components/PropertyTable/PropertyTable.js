import React from 'react';
import PropTypes from 'prop-types';
import testImage from './test.png';
import styles from './styles.scss';

const PropertyTable = ({ tableItems : { temp, visibility, windSpeed, }, }) => {
    return (
        <div className="property-table">
            <div className="property-table-item temp-item">Current temperature: {temp}</div>
            <div className="property-table-item visibility-item">Current visibility: {visibility}</div>
            <div className="property-table-item wind-item">Current speed of wind: {windSpeed}</div>
            <img src={testImage} alt='testImage' />
        </div>
    );

};

PropertyTable.propTypes = {
    tableItems: PropTypes.shape({
        temp: PropTypes.number,
        visibility: PropTypes.number,
        windSpeed: PropTypes.number,
    }),
};

export default PropertyTable;
