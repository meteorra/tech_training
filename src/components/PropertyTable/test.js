import React from 'react';
import PropertyTable from './PropertyTable';
import renderer from 'react-test-renderer';

function setup() {
    const props = {
        tableItems: {
            temp: 20,
            visibility: 100,
            windSpeed: 1000,
        },
    };

    const render = renderer.create(<PropertyTable {...props} />);

    return {
        props,
        render,
    };
}

describe('PropertyTable component', () => {

    it('renders correctly certain values of temp, visibility & windSpeed', () => {
        const { render, } = setup();

        expect(render.toJSON()).toMatchSnapshot();
    });

});

