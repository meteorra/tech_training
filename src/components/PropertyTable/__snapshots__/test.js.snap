// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyTable component renders correctly certain values of temp, visibility & windSpeed 1`] = `
<div
  className="property-table"
>
  <div
    className="property-table-item temp-item"
  >
    Current temperature: 
    20
  </div>
  <div
    className="property-table-item visibility-item"
  >
    Current visibility: 
    100
  </div>
  <div
    className="property-table-item wind-item"
  >
    Current speed of wind: 
    1000
  </div>
</div>
`;
